pkg/syndesis/action/attach.go

```diff
@@ -0,0 +1,161 @@
+package action
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/configuration"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Attach a detached Syndesis installation to the Syndesis custom resource
+type Attach struct {}
+
+
+func (a *Attach) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusAttaching)
+}
+
+func (a *Attach) Execute(syndesis *v1alpha1.Syndesis) error {
+
+	// Checking that there's only one installation to avoid stealing resources
+	if anotherInstallation, err := isAnotherActiveInstallationPresent(syndesis); err != nil {
+		return err
+	} else if anotherInstallation {
+		target := syndesis.DeepCopy()
+
+		target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusNotInstalled
+		target.Status.Reason = v1alpha1.SyndesisStatusReasonDuplicate
+		target.Status.Description = "Cannot merge because another Syndesis resources is present in the same namespace"
+
+		logrus.Error("Cannot merge resource ", syndesis.Name, " because another Syndesis resources is present in the same namespace")
+		return sdk.Update(target)
+	}
+
+	logrus.Info("Attaching Syndesis installation to resource ", syndesis.Name)
+
+	for _, selector := range getAllManagerSelectors() {
+		for _, metaType := range getAllManagedResourceTypes() {
+
+			options := sdk.WithListOptions(&selector)
+			list := metav1.List{
+				TypeMeta: metaType,
+			}
+			if err := sdk.List(syndesis.Namespace, &list, options); err != nil {
+				return err
+			}
+
+			for _, obj := range list.Items {
+				res, err := util.LoadKubernetesResource(obj.Raw)
+				if err != nil {
+					return err
+				}
+				setNamespaceAndOwnerReference(res, syndesis)
+				if err := sdk.Update(res); err != nil {
+					return err
+				}
+			}
+
+		}
+	}
+
+	syndesisVersion, err := configuration.GetSyndesisVersionFromNamespace(syndesis.Namespace)
+	if err != nil {
+		return err
+	}
+
+	target := syndesis.DeepCopy()
+	target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusStarting
+	target.Status.Reason = v1alpha1.SyndesisStatusReasonMissing
+	target.Status.Description = ""
+	target.Status.Version = syndesisVersion
+
+	logrus.Info("Syndesis installation attached to resource ", syndesis.Name)
+	return sdk.Update(target)
+}
+
+func getAllManagerSelectors() []metav1.ListOptions {
+	return []metav1.ListOptions {
+		{
+			LabelSelector: "syndesis.io/app=syndesis,syndesis.io/type=infrastructure",
```

Comment:
actually I would thought that this selector should be enough. Do you know which other objects which neede to be attached requires special label ?

IMO it should be enough to select on "app=syndesis", and then on any object for doing that kind of attachment.

'happy to fix that on the template side.

Reply:
For the TODO, the syndesis.io/app is different and there's also a error with the DB PVC not belonging to the "infrastructure" type. I've fixed the template but I expect existing installations not to be fixed...